From b0f7dcdf1621515b9cb3be89da8a0dcd33a48180 Mon Sep 17 00:00:00 2001
From: Mats Erik Andersson <gnu@gisladisker.se>
Date: Mon, 23 May 2016 23:06:47 +0200
Subject: [PATCH 24/42] hostname: Avoid a trailing space.

Suppress the space character after the last presented IP number.
---
 ChangeLog      | 10 ++++++++++
 src/hostname.c |  5 ++++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/src/hostname.c b/src/hostname.c
index e18b9cc8..f2b587db 100644
--- a/src/hostname.c
+++ b/src/hostname.c
@@ -337,7 +337,10 @@ get_ip_addresses (const char *const host_name)
             }
 
           strcat (addresses, address);
-          strcat (addresses, " ");
+
+	  /* Insert a separating space character.  */
+	  if (ht->h_addr_list[i+1] != NULL)
+	    strcat (addresses, " ");
         }
     }
 
-- 
2.20.1.791.gb4d0f1c61a

